engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1828,6 +1796,104 @@ public void run()
     return rsp;
   }
 
+  static class UpdateOperatorLatencyContext
+  {
+    StreamingContainerManager scm;
```

Comment:
Why depend on scm for the context? It needs rpcLatencies, endWindowStatsOperatorMap and a few other things but not everything else. We should try to decompose this monolithic class into pieces that can be developed and tested independently. For the latency test, I would expect that no instance of scm is needed.